ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -298,6 +373,18 @@ func (r *RayClusterReconciler) reconcilePods(instance *rayiov1alpha1.RayCluster)
 				return err
 			}
 		}
+	} else {
+		// we have exactly one head pod running
+		if headPods.Items[0].Annotations != nil {
```

Comment:
do we still manually delete head node? Will liveness restart the pod auto?

Reply:
liveness probe will delete it. here the code is for readiness probe. we can do other operations rather than kill the pod. this can be a future implmentation.

Reply:
So now we have 2 policies. One is liveness and one is RayCluster controller to both delete the pod?

Reply:
yeah, but the readiness probe failures can be updated so that we can do a better control in the future.